sycl/test-e2e/OptionalKernelFeatures/throw-exception-for-out-of-registers-on-kernel-launch.cpp

```diff
@@ -92,11 +94,13 @@ int main() {
              output[i] = vadd;
              output[i] += sum;
            });
-     }).wait();
-  } catch (sycl::exception &e) {
+     }).wait_and_throw();
+  } catch (const sycl::exception &e) {
     using std::string_view_literals::operator""sv;
     auto Msg = "Exceeded the number of registers available on the hardware."sv;
-    if (std::string(e.what()).find(Msg) != std::string::npos) {
+    auto Errc = sycl::make_error_code(sycl::errc::nd_range);
```

Comment:
@GeorgeWeb, I was looking at the PR and got confused by it. Could you please clarify what is the key change here? PR description says that we used to display confusing error to users, but check for the exception message wasn't changed by the PR.

What was the confusing part then? I see that in #12363 we had a bug report which contains invalid message, but this test had been introduced a year ago before that issue was submitted in #9106. What am I missing here?

Reply:
In summary the wrong part of the message as per the PRs description was just the plugin error code description - `PI_ERROR_INVALID_WORK_GROUP_SIZE` should have been `PI_ERROR_OUT_OF_LAUNCH_RESOURCES`. The reason I've added the `errc::nd_range` error code check was just for more verbosity but it wasn't of importance here.

---

The real issue about "reporting a completely wrong message" in the OPs report (https://github.com/intel/llvm/issues/12363) was due to a mistake on this line https://github.com/intel/llvm/pull/9106/files#diff-7525901710934f7bdb2ad36238c4b67163f112d3bd233db7af0b0078b5b01e80R3263 which was fixed by this UR cuda change https://github.com/oneapi-src/unified-runtime/pull/1299